apsuite/optics_analysis/base_correction.py

```diff
@@ -0,0 +1,188 @@
+"""."""
+
+from collections import namedtuple as _namedtuple
+from copy import deepcopy as _dcopy
+import numpy as np
+
+__KNOBTYPES = _namedtuple('KnobTypes', ['Focusing', 'Defocusing'])
+
+
+class KnobTypes(__KNOBTYPES):
+    """."""
+
+    @property
+    def ALL(self):
+        """."""
+        return self.Focusing + self.Defocusing
+
+
+class BaseCorr():
+    """."""
+
+    METHODS = _namedtuple('Methods', ['Additional', 'Proportional'])(0, 1)
+    GROUPING = _namedtuple('Grouping', ['Individual', 'TwoKnobs'])(0, 1)
+    CORR_STATUS = _namedtuple('CorrStatus', ['Fail', 'Sucess'])(0, 1)
+
+    def __init__(self, model=None, acc=None, method=None, grouping=None):
+        """."""
+        self.model = model
+        self.acc = acc
+        self._method = BaseCorr.METHODS.Proportional
+        self._grouping = BaseCorr.GROUPING.TwoKnobs
+        self.knobs = None
+        self.fam = None
+        self.strength_type = None
```

Comment:
I think it would be better if `strength_type` were class attribute, not an object attribute!

Reply:
ok!